### PR TITLE
Allow newlines in switch statement

### LINKED ISSUE #206: Newlines or not to newline ;-)
@kaikreuzer can/should we allow newlines in the following case:

```java
        switch (channelId) {
            case CHANNEL_WIND_DIRECTION:
                return new DecimalType(windDirection);

            case CHANNEL_AVG_WIND_SPEED:
                return new DecimalType(avgWindSpeed);

            case CHANNEL_WIND_SPEED:
                return new DecimalType(windSpeed);

            case CHANNEL_TEMPERATURE:
                return new DecimalType(temperature);

            case CHANNEL_CHILL_TEMPERATURE:
                return new DecimalType(chillTemperature);

            default:
                return super.convertToState(channelId);
        }
```

I think this is quite okay and will not become more readable by removing newlines, they add value here, but:

```
checkstyle | 3 | 172 | style | NoEmptyLineSeparatorCheck | Remove empty line after opening brace
checkstyle | 3 | 176 | style | NoEmptyLineSeparatorCheck | Remove empty line before closing brace
checkstyle | 3 | 179 | style | NoEmptyLineSeparatorCheck | Remove empty line before closing brace
checkstyle | 3 | 182 | style | NoEmptyLineSeparatorCheck | Remove empty line before closing brace
checkstyle | 3 | 185 | style | NoEmptyLineSeparatorCheck | Remove empty line before closing brace
checkstyle | 3 | 188 | style | NoEmptyLineSeparatorCheck | Remove empty line before closing brace
```

PS: there is not even a closing 'brace' in this

### PR DESCRIPTION
**Pull request by tanyatanyatanya**
Made some changes in NoEmptyLineSeparatorCheck to allow newlines in some cases for switch statement.
Fixed #206

### CODE REVIEW

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java [hunk 1]
```diff
@@ -62,11 +62,17 @@
...
  * @author Svilen Valkanov - Initial contribution
+ * @author Tanya Georgieva - Added check for switch statement without braces
  */
 public class NoEmptyLineSeparatorCheck extends AbstractCheck {
 
     private static final String MSG_LINE_AFTER_OPENING_BRACE_EMPTY = "Remove empty line after opening brace";
     private static final String MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY = "Remove empty line before closing brace";
+    private static final String MSG_LINE_FOR_CASE_WITHOUT_BRACES_EMPTY = "Remove empty line";
+
+    private boolean isCaseGroupWithoutBraces = false;
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java]: Note that false is default if you would have left them out here.

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java [hunk 2]
```diff
@@ -62,11 +62,17 @@
  * </ul>
  *
  * @author Svilen Valkanov - Initial contribution
+ * @author Tanya Georgieva - Added check for switch statement without braces
  */
 public class NoEmptyLineSeparatorCheck extends AbstractCheck {
 
     private static final String MSG_LINE_AFTER_OPENING_BRACE_EMPTY = "Remove empty line after opening brace";
     private static final String MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY = "Remove empty line before closing brace";
+    private static final String MSG_LINE_FOR_CASE_WITHOUT_BRACES_EMPTY = "Remove empty line";
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java]: You have a really generic message maybe you should give the constant a generic name as well?

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java [hunk 3]
```diff
@@ -118,15 +124,68 @@ public void visitToken(DetailAST ast) {
...
+                     * and we should not log warning message
+                     *
+                     * case 1:
+                     * ....return number;
+                     * (empty line)
+                     */
+                    boolean isEmptyLineWithWarningForCaseGroup = !(isLineAfterOpeningBrace
+                            && (isCaseGroupWithBraces || isCaseGroupWithoutListOfStatements));
+                    /*
+                     * if the case clause is without braces the empty line could be after the colon(:)
```
Comment (martinvw) [inline custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java]: Really clear 👍

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java [hunk 4]
```diff
@@ -18,17 +18,82 @@
...
+ * @author Svilen Valkanov - Initial contribution
+ * @author Tanya Georgieva - Added tests for switch statement without braces
  */
 public class NoEmptyLineSeparatorCheckTest extends AbstractStaticCheckTest {
 
     private static final String TEST_DIRECTORY_NAME = "noEmptyLineSeparatorCheck";
 
     private static final String MSG_LINE_AFTER_OPENING_BRACE_EMPTY = "Remove empty line after opening brace";
     private static final String MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY = "Remove empty line before closing brace";
+    private static final String MSG_LINE_FOR_CASE_WITHOUT_BRACES_EMPTY = "Remove empty line";
```
Comment (martinvw) [inline custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: See above

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java [hunk 5]
```diff
@@ -18,17 +18,82 @@
...
 
     private static DefaultConfiguration config = createCheckConfig(NoEmptyLineSeparatorCheck.class);
 
+    @Test
+    public void verifyValidSwitchDefitionWithoutBraces() throws Exception {
+        verifyJavaFileNoErrors("ValidSwitchDefitionWithoutBraces.java");
```
Comment (martinvw) [inline custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: Typo in 'Defition'

File: custom-checks/checkstyle/src/test/resources/checkstyle/noEmptyLineSeparatorCheck/EmptyLineAfterInvalidDefaultCaseWithoutBraces.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+public class EmptyLineAfterInvalidDefaultCaseWithoutBraces {
+
+    // empty line after class is allowed
+
+    public void method() {
+        switch (channelId) {
+            case CHANNEL_WIND_DIRECTION:
+                return new DecimalType(windDirection);
+            case CHANNEL_AVG_WIND_SPEED:
```
Comment (martinvw) [inline custom-checks/checkstyle/src/test/resources/checkstyle/noEmptyLineSeparatorCheck/EmptyLineAfterInvalidDefaultCaseWithoutBraces.java]: Maybe add a newline here, I believe this is not the focus point of this test, is it?

File: sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml [hunk 7]
```diff
@@ -195,7 +195,7 @@
     <module name="org.openhab.tools.analysis.checkstyle.AvoidScheduleAtFixedRateCheck">
         <property name="severity" value="warning" />
     </module>
-
+
```
Comment (martinvw) [inline sat-plugin/src/main/resources/rulesets/checkstyle/rules.xml]: This does not look intended / desired?

**Comment by martinvw** [comment]
If those small remarks are handled LGTM

File: custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java (hunk 2 above)
Comment (tanyatanyatanya) [reply custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/NoEmptyLineSeparatorCheck.java]: Yes, I agree. I will change it.

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java (hunk 5 above)
Comment (tanyatanyatanya) [reply custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java]: Fixed. Thanks.

File: custom-checks/checkstyle/src/test/resources/checkstyle/noEmptyLineSeparatorCheck/EmptyLineAfterInvalidDefaultCaseWithoutBraces.java (hunk 6 above)
Comment (tanyatanyatanya) [reply custom-checks/checkstyle/src/test/resources/checkstyle/noEmptyLineSeparatorCheck/EmptyLineAfterInvalidDefaultCaseWithoutBraces.java]: I added a new line there and checked the test.

**Comment by tanyatanyatanya** [comment]
I fixed the comments. About the - private boolean isCaseGroupWithoutBraces = false; I want them to be false by default and I thought would be more clear with equals to false.

**Comment by martinvw** [comment]
LGTM

**Comment by svilenvul** [comment]
Thanks all !

**Pull request merged**